Monika-After-Story/MonikaModDev
# Affection-based ily and ily2

There are two parts to this, the first being the `I love you!` topic now being affection based with the second being the change of the main menu `I love you` prompt being changed to `I love you, too` if the previous topic had Monika saying some variant of `I love you` near its end.

# Affection-based ily

- `monika_love` now has affection dependent paths. Normal+ will use the same code that is currently used, with lower affections now having new responses. Upset has been broken into 2 parts, `>-50` and `<=-50` to allow for different dialogue depending on how upset she is. The upper level still shows hope for the relationship, whereas the lower level has her more cynical, not sure how much of it is just lip-service. Distressed has her in complete doubt that you mean it, while Broken gives no response other than `"..."`

- Another change is `I love you!` is now only available on the main menu if affection is `>-50`. This makes it so the player is more likely to mean it if they say it, plus is another sign that Monika does not feel loved.

- Some minor coding changes are also present, including forcing the counter to increase if at a milestone for Normal+ (ie multiples of 5) no matter how long it's been since the last ily so as to prevent getting the dialogue multiple times in a row, as well as how these milestones are checked, resulting in not having to reset the counter once it reaches 50.

- The first time clicking `I love you!` dialogue has been changed. Before it implied it was the first time you ever told her you loved her, but to play the mod at all, you have to tell her you love her in the introduction. The old dialogue has been slightly reworked and move to the introduction as part of a timed menu where you can tell her you love her after she says she loves you, but before she asks. If you miss the answer, then she asks and you get the old intro dialogue. The first time clicking `I love you` if you miss the timed response now reflects that it's the first time you said it without being put on the spot.

- A related change to the intro has also been made where if you reject her love and get the scare, you can only get the scare once, and any subsequent load will lead to her asking if you love her with only a `Yes` response. You either answer yes or you don't play the mod. Having the scare over and over again loses its effectiveness.

## Testing

- verify that at Normal+ affection you get the usual `ily` dialogue.

- verify at Upset, `>-50` you get the hopeful Upset dialogue and at `<-50` you get the cynical Upset dialogue.

- verify at Distressed and Broken you get the corresponding dialogue.

- verify at any affection `<=-50` that `I love you!` is not present on the main menu.

- verify if it's been more than 3 minutes after the last `ily` that the appropriate affection is given and that `persistent._mas_monika_lovecounter` increments.

- verify if it's been less than 3 minutes, no affection is given and `persistent._mas_monika_lovecounter` only increments if at a milestone for Normal+ (ie it never gets stuck on an increment of 5).

- on a fresh persistent, check the intro and verify that the timed menu is present for a reasonable time and that if clicked `persistent._mas_monika_lovecounter` increments and that a subsequent `ily` does not give first time dialogue.

- also on a fresh persistent, check the intro and verify if you reject her love, the next load only has a `Yes` option when she asks if you love her.

# I love you, too!
This creates the return key `"love"` that is added to topics that have Monika telling the player that she loves them near the end of the topic. When this key is reached, it changes the `I love you!` prompt on the main menu to `I love you, too!` for 10 seconds and then has Monika react with a cute expression. You can read why in #2051. This also gives the same affection as `I love you!` as well as shares the same 3 minute timer. This is present on select topics, greets, moods, compliments and assorted other events. Check the code for the complete list.

## Testing

- verify any path that returns `"love"` changes the `I love you!` prompt on the main menu to `I love you, too!` for 10 seconds. Note: if you click the main menu and leave it up, the prompt will stay there until you leave that menu.

- verify I didn't screw any paths up by adding returns that prevent us from reaching important code. (srsly, do this).

- verify that if you click `I love you, too!` you get one of the correct expressions with no dialogue. the expression options for Normal+ and Enam+ affection can be seen in `sprite-chart.rpy` under `image monika ATL_love_too_norm_plus` and `image monika ATL_love_too_enam_plus` respectively. For the upset path (`>-50` since this follows the same rule as `I love you!` for showing on the main menu) is `2eka`.
